Utilities.cs

```diff
@@ -0,0 +1,236 @@
+using System.Threading.Tasks;
+using TrainingProject.General;
+using TrainingProject.ProductMangment;
+
+namespace TrainingProject
+{
+    internal class Utilities
+    {
+        private static List<Product> ProductsList = new();
+        private static Inventory inventory;
+
+        internal static void InitializeStock()
+        {
+            Product p1 = new Product("Sugar", new Price() { ItemPrice = 10, Currency = Currency.USD }, 100);
+            Product p2 = new Product("Cake decorations", new Price() { ItemPrice = 8, Currency = Currency.NIS }, 20);
+            Product p3 = new Product("Strawberry", new Price() { ItemPrice = 3, Currency = Currency.JOD }, 10);
+            ProductsList.Add(p1);
+            ProductsList.Add(p2);
+            ProductsList.Add(p3);
+            inventory = new Inventory(ProductsList);
+        }
+
+        internal static void ShowMainMenu()
+        {
+            Console.ResetColor();
+            Console.Clear();
+            Console.WriteLine("********************");
+            Console.WriteLine("* Select an action *");
+            Console.WriteLine("********************");
+
+            Console.WriteLine("1: Add Product");
+            Console.WriteLine("2: View all products");
+            Console.WriteLine("3: Edit a product");
+            Console.WriteLine("4: Delete a product");
+            Console.WriteLine("5: Search for a product");
+            Console.WriteLine("0: Close application");
+
+            Console.Write("Your selection: ");
+
+            string? userSelection = Console.ReadLine();
+            switch (userSelection)
+            {
+                case "1":
+                    AddProduct();
+                    break;
+                case "2":
+                    ViewAllProducts();
+                    break;
+                case "3":
+                    EditProductOption();
+                    break;
+                case "4":
+                    DeleteProductOption();
+                    break;
+                case "5":
+                    SearchForProductOption();
+                    break;
+                case "0":
+                    break;
+                default:
+                    Console.WriteLine("Invalid selection. Please try again.");
+                    break;
+            }
+
+        }
+        private static void AddProduct()
+        {
+            Console.Write("Enter Product Name: ");
+            string? ProductName = Console.ReadLine();
+
+            Console.Write("Enter Product Price: ");
+            double ProductItemPrice = double.Parse(Console.ReadLine());
+
+            Console.Write("Enter Product Currency: ");
+            string? ProductItemCurrency = Console.ReadLine().ToUpper();
+
+            Console.Write("Enter Product Quantity: ");
+            int ProductQuantity = int.Parse(Console.ReadLine());
+
+            Currency CurrencyValue;
+            Enum.TryParse(ProductItemCurrency, out CurrencyValue);
```

Comment:
1. no need for declaration
2. camelCase variable naming convention
3. you should check if tryParse return false, then decide which is the default value you should assign
```suggestion
      
            Enum.TryParse(ProductItemCurrency, out var currencyValue);
```